exchange/bilaxy/exchange.go

```diff
@@ -0,0 +1,310 @@
+}
+
+func (e *Bilaxy) SetCoinConstraint(coinConstraint *exchange.CoinConstraint) {
+	coinConstraintMap.Set(fmt.Sprintf("%d", coinConstraint.CoinID), coinConstraint)
+}
+
+func (e *Bilaxy) GetCoins() []*coin.Coin {
+	coinList := []*coin.Coin{}
+	keySort := []int{}
+	for _, key := range coinConstraintMap.Keys() {
+		id, _ := strconv.Atoi(key)
+		keySort = append(keySort, id)
+	}
+	sort.Ints(keySort)
+	for _, key := range keySort {
+		c := coin.GetCoinByID(key)
+		if c != nil {
+			coinList = append(coinList, c)
+		}
+	}
+	return coinList
+}
+
+func (e *Bilaxy) GetCoinBySymbol(symbol string) *coin.Coin {
+	for _, id := range coinConstraintMap.Keys() {
+		if tmp, ok := coinConstraintMap.Get(id); ok {
+			cc := tmp.(*exchange.CoinConstraint)
+			if cc.ExSymbol == symbol {
+				return cc.Coin
+			}
+		} else {
+			log.Printf("Get ID %s CoinConstraint Err", id)
+		}
+	}
+	return nil
+}
+
+func (e *Bilaxy) GetSymbolByCoin(coin *coin.Coin) string {
+	key := fmt.Sprintf("%d", coin.ID)
+	if tmp, ok := coinConstraintMap.Get(key); ok {
+		cc := tmp.(*exchange.CoinConstraint)
+		return cc.ExSymbol
+	}
+	return ""
+}
+
+func (e *Bilaxy) DeleteCoin(coin *coin.Coin) {
+	coinConstraintMap.Remove(fmt.Sprintf("%d", coin.ID))
+}
+
+/*************** Pairs on the Exchanges ***************/
+func (e *Bilaxy) GetPairConstraint(pair *pair.Pair) *exchange.PairConstraint {
+	if tmp, ok := pairConstraintMap.Get(fmt.Sprintf("%d", pair.ID)); ok {
+		return tmp.(*exchange.PairConstraint)
+	}
+	return nil
+}
+
+func (e *Bilaxy) SetPairConstraint(pairConstraint *exchange.PairConstraint) {
+	pairConstraintMap.Set(fmt.Sprintf("%d", pairConstraint.PairID), pairConstraint)
+}
+
+func (e *Bilaxy) GetPairs() []*pair.Pair {
+	pairList := []*pair.Pair{}
+	keySort := []int{}
+	for _, key := range pairConstraintMap.Keys() {
+		id, _ := strconv.Atoi(key)
+		keySort = append(keySort, id)
+	}
+	sort.Ints(keySort)
+	for _, key := range keySort {
+		p := pair.GetPairByID(key)
+		if p != nil {
+			pairList = append(pairList, p)
+		}
+	}
+	return pairList
+}
+
+func (e *Bilaxy) GetPairBySymbol(symbol string) *pair.Pair {
+	for _, id := range pairConstraintMap.Keys() {
+		if tmp, ok := pairConstraintMap.Get(id); ok {
+			pc := tmp.(*exchange.PairConstraint)
+			if pc.ExSymbol == symbol {
+				return pc.Pair
+			}
+		}
+	}
+	return nil
+}
+
+func (e *Bilaxy) GetSymbolByPair(pair *pair.Pair) string {
+	pairConstraint := e.GetPairConstraint(pair)
+	if pairConstraint != nil {
+		return pairConstraint.ExSymbol
+	}
+	return ""
+}
+
+func (e *Bilaxy) HasPair(pair *pair.Pair) bool {
+	return pairConstraintMap.Has(fmt.Sprintf("%d", pair.ID))
+}
+
+func (e *Bilaxy) DeletePair(pair *pair.Pair) {
+	pairConstraintMap.Remove(fmt.Sprintf("%d", pair.ID))
+}
+
+/**************** Exchange Constraint ****************/
+func (e *Bilaxy) GetConstraintFetchMethod(pair *pair.Pair) *exchange.ConstrainFetchMethod {
+	constrainFetchMethod := &exchange.ConstrainFetchMethod{}
+	constrainFetchMethod.PublicAPI = false
```

Comment:
set this to true if coin, pair and orderbook method are successfully implemented.